Nama : Dicki Pramana Putra  
NIM : 250401020112  
Kelas : IF-207  

In [ ]:
import pandas as pd
import numpy as np
from scipy.stats.mstats import winsorize

df = pd.read_csv("/housing_dirty.csv")
print("Dataset berhasil dimuat!")
print("Shape:", df.shape)
print(df.isnull().sum())
df.info()          # cek tipe data & missing
df.describe()      # statistik dasar
df.isnull().sum()  # jumlah missing per kolom
print("cek duplikat:", df.duplicated().sum())

Dataset berhasil dimuat!
Shape: (130, 7)
id               0
luas_m2         18
harga_juta      17
kota             0
kamar           10
tahun_bangun     0
kondisi          0
dtype: int64
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 130 entries, 0 to 129
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id            130 non-null    int64  
 1   luas_m2       112 non-null    float64
 2   harga_juta    113 non-null    float64
 3   kota          130 non-null    object 
 4   kamar         120 non-null    float64
 5   tahun_bangun  130 non-null    int64  
 6   kondisi       130 non-null    object 
dtypes: float64(3), int64(2), object(2)
memory usage: 7.2+ KB
cek duplikat: 0


In [ ]:
df.drop_duplicates(inplace=True)
print('Setelah hapus duplikat:', df.shape)

Setelah hapus duplikat: (130, 7)


In [ ]:
df['kota']    = df['kota'].str.strip().str.title()
df['kondisi'] = df['kondisi'].str.strip().str.lower()

In [ ]:
df['luas_m2']    = df['luas_m2'].fillna(df['luas_m2'].median())
df['harga_juta'] = df['harga_juta'].fillna(df['harga_juta'].median())
df['kamar']      = df['kamar'].fillna(df['kamar'].mode()[0])

In [ ]:
for col in ['harga_juta', 'luas_m2', 'tahun_bangun']:
    Q1, Q3 = df[col].quantile([0.25, 0.75])
    IQR = Q3 - Q1
    df[col] = df[col].clip(Q1-1.5*IQR, Q3+1.5*IQR)

In [ ]:
# STEP 5 — Validasi & Ekspor
assert df.isnull().sum().sum() == 0, 'Masih ada missing!'
assert df.duplicated().sum() == 0, 'Masih ada duplikat!'
print('Shape akhir:', df.shape)
df.to_csv('housing_clean.csv', index=False)
print('Dataset bersih tersimpan!')

Shape akhir: (130, 7)
Dataset bersih tersimpan!


**Apa yang dipelajari:**
Pertemuan ini mempelajari proses data cleaning meliputi penanganan missing values,
duplikat, standardisasi teks, dan penanganan outlier menggunakan metode IQR clipping
pada dataset properti rumah.

**Temuan utama:**
Dataset awal memiliki 130 baris dengan missing values pada kolom luas_m2 (18),
harga_juta (17), dan kamar (10). Setelah proses cleaning, semua missing values
berhasil ditangani dan dataset bersih tersimpan sebagai housing_clean.csv.
